### PR TITLE
Migrate to Java 11

### LINKED ISSUE #2944: Broker should be build with the latest Java LTS (11) version
**Is your feature request related to a problem? Please describe.**

Currently the broker only supports Java 8. 
To be competitive and able to use new language features, bug fixes, performance improvements etc we should use newer java versions.

**Describe the solution you'd like**

As a first step we want to support the latest LTS version, which is Java 11.
Later we can decided if we want to  support non-LTS versions.

**Additional context**

The scheduler and the dispatcher are currently using `sun.misc.Unsafe`.
As an easy fix we could do this similar to agrona/aeron, which simply enables the internal modules on the build process. 

Fixing these issues can be tackled later when the JDK provides an alternative to that.

### PR DESCRIPTION
**Pull request by ChrisKujawa**
## Description

 * removed the java enforcer plugin, which enforces java 8
 * in order to use java 11 and the maven dependency plugin we
 have to use a higher dependency for the `maven-dependency-analyzer`
 * added javax as dependency, since this is not build in anymore
 * updated CI to java 11
   *  remove unused and deprected java option
      * cgroup memory limit was deprecated in Java 11
        https://medium.com/adorsys/jvm-memory-settings-in-a-container-environment-64b0840e1d9e
      * experimental options was unused after removing the other option

  * replace deprecated class usage
    * fix jdk nashorn usage - use jackson instead

 *  fix spotbugs on java 11
    * Spotbugs finds redudant null check because java 11 compiler generates
     an null check in the bytecode, which is then found by spotbugs
    * ignore redudant null check for now cause of spotbugs/spotbugs#756
   * use java 8 for client
     * bpmn-model, gateway-protocol-impl and client is build with java 8

## Related issues

closes #2944

## Pull Request Checklist

- [X] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [X] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [X] If submitting code, please run `mvn clean install -DskipTests` locally before committing

### CODE REVIEW

File: parent/pom.xml [hunk 1]
```diff
@@ -546,8 +557,11 @@
           <artifactId>maven-compiler-plugin</artifactId>
           <version>${plugin.version.compiler}</version>
           <configuration>
-            <source>1.8</source>
-            <target>1.8</target>
+            <release>11</release>
+
+            <!-- WE NEED THIS FOR IDEA Intellij -->
+            <source>11</source>
+            <target>11</target>
```
Comment (saig0) [inline parent/pom.xml]: The compiler plugin is already configured In the parent pom `camunda-release-parent` like:

```
<plugin>
          <groupId>org.apache.maven.plugins</groupId>
          <artifactId>maven-compiler-plugin</artifactId>
          <version>${plugin.version.compiler}</version>
          <configuration>
            <source>${version.java}</source>
            <target>${version.java}</target>
          </configuration>
        </plugin>
```

So, we just need to change the property `version.java` and setting the release version.

**Review by saig0 — CHANGES_REQUESTED** [review]
:tada:

File: parent/pom.xml (hunk 1 above)
Comment (Zelldon) [reply parent/pom.xml]: Please see latest commit

**Review by saig0 — APPROVED** [review]

**Pull request merged**